src/seedu/addressbook/commands/SortCommand.java

```diff
@@ -0,0 +1,152 @@
+
+                for (int i = 0; i < non_sortable.size(); i++) {
+                    try {
+                        addressBook.addPerson(new Person(non_sortable.get(i)));
+                    }
+                    catch (DuplicateDataException e) {
+                        System.out.println("It wont happen anyway");
+                    }
+                }
+
+                for (int i = 0; i < sortable.size(); i++) {
+                    try {
+                        addressBook.addPerson(new Person(sortable.get(i)));
+                    }
+                    catch (DuplicateDataException e) {
+                        System.out.println("It wont happen anyway");
+                    }
+                }
+
+                break;
+            }
+            case " email": {
+                ArrayList<ReadOnlyPerson> sortable = new ArrayList<>();
+                ArrayList<ReadOnlyPerson> non_sortable = new ArrayList<>();
+
+                for (int i = 0; i < temp_allPersons.size(); i++) {
+                    if (temp_allPersons.get(i).getEmail().isPrivate()) non_sortable.add(temp_allPersons.get(i));
+                    else sortable.add(temp_allPersons.get(i));
+                }
+
+                Collections.sort(non_sortable, new Comparator<ReadOnlyPerson>() {
+                    @Override
+                    public int compare(ReadOnlyPerson o1, ReadOnlyPerson o2) {
+                        return o1.getName().toString().compareToIgnoreCase(o2.getName().toString());
+                    }
+                });
+
+                Collections.sort(sortable, new Comparator<ReadOnlyPerson>() {
+                    @Override
+                    public int compare(ReadOnlyPerson o1, ReadOnlyPerson o2) {
+                        return o1.getEmail().toString().compareToIgnoreCase(o2.getEmail().toString());
+                    }
+                });
+
+                addressBook.clear();
+
+                for (int i = 0; i < non_sortable.size(); i++) {
+                    try {
+                        addressBook.addPerson(new Person(non_sortable.get(i)));
+                    }
+                    catch (DuplicateDataException e) {
+                        System.out.println("It wont happen anyway");
+                    }
+                }
+
+                for (int i = 0; i < sortable.size(); i++) {
+                    try {
+                        addressBook.addPerson(new Person(sortable.get(i)));
+                    }
+                    catch (DuplicateDataException e) {
+                        System.out.println("It wont happen anyway");
+                    }
+                }
+
+                break;
+            }
+            default:
+                return new CommandResult(MESSAGE_WRONG_ARGUMENT);
+        }
+
+        temp_allPersons = addressBook.getAllPersons().immutableListView();
+        if (argument.equals(" name")) return new CommandResult("Sorted by name", temp_allPersons);
```

Comment:
The conditional should be placed on a separate line according to our [Coding Standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#conditionals).